src/Workspaces/Core/Portable/SymbolKey/SymbolKey.AnonymousTypeSymbolKey.cs

```diff
@@ -15,19 +15,25 @@ public static void Create(INamedTypeSymbol symbol, SymbolKeyWriter visitor)
             {
                 Debug.Assert(symbol.IsAnonymousType);
 
-                var properties = symbol.GetMembers().OfType<IPropertySymbol>().ToArray();
-                var propertyTypes = properties.Select(p => p.Type).ToImmutableArray();
-                var propertyNames = properties.Select(p => p.Name).ToImmutableArray();
+                var properties = symbol.GetMembers().OfType<IPropertySymbol>().ToImmutableArray();
+                var propertyTypes = properties.SelectAsArray(p => p.Type);
+                var propertyNames = properties.SelectAsArray(p => p.Name);
+                var propertyIsReadOnly = properties.SelectAsArray(p => p.SetMethod == null);
+                var propertyLocations = properties.SelectAsArray(p => p.Locations.FirstOrDefault());
 
                 visitor.WriteSymbolKeyArray(propertyTypes);
                 visitor.WriteStringArray(propertyNames);
+                visitor.WriteBooleanArray(propertyIsReadOnly);
+                visitor.WriteLocationArray(propertyLocations);
             }
 
             public static int GetHashCode(GetHashCodeReader reader)
             {
                 // The hash of the underlying type is good enough, we don't need to include names.
                 var symbolKeyHashCode = reader.ReadSymbolKeyArrayHashCode();
-                var elementNames = reader.ReadStringArray();
+                var propertyNames = reader.ReadStringArray();
+                var propertyIsReadOnly = reader.ReadBooleanArray();
```

Comment:
Note: all the GetHashCode functionality was removed in: https://github.com/dotnet/roslyn/pull/14330

However, to answer your question:

SymbolKeys are just strings.  Therefore, if we're computing anything about them, we need to stream over all the data they have.  If we don't read the data, then the stream-pointer will be in teh wrong place, and that will mean we can't properly read the rest of the data we need.